minecraft/src/main/java/net/fabricmc/loader/impl/game/minecraft/MinecraftGameProvider.java

```diff
@@ -243,7 +243,7 @@ public boolean locateGame(FabricLauncher launcher, String[] args) {
 		// expose obfuscated jar locations for mods to more easily remap code from obfuscated to intermediary
 		ObjectShare share = FabricLoaderImpl.INSTANCE.getObjectShare();
 		share.put("fabric-loader:inputGameJar", gameJars.get(0)); // deprecated
-		share.put("fabric-loader:inputGameJars", gameJars);
+		share.put("fabric-loader:inputGameJars", new ArrayList<>(gameJars));
```

Comment:
I would use unmodifiableList

Reply:
wrapping gameJars with Collections.unmodifiableList doesn't fix the issue that the underlying list is mutated in place. I could make the new list unmodifiable if that's what you mean.